src/Identity/Controllers/SsoController.cs

```diff
@@ -60,13 +64,16 @@ public async Task<IActionResult> PreValidate(string domainHint)
                 var requestPath = $"/Account/PreValidate?domainHint={domainHint}&culture={culture}";
                 var httpClient = _clientFactory.CreateClient("InternalSso");
                 using var responseMessage = await httpClient.GetAsync(requestPath);
+                var responseJson = await responseMessage.Content.ReadAsStringAsync();
+
                 if (responseMessage.IsSuccessStatusCode)
                 {
-                    // All is good!
-                    return new EmptyResult();
+                    var obj = JsonConvert.DeserializeObject<ExpandoObject>(responseJson, new ExpandoObjectConverter());
+                    return Json(obj);
                 }
+
                 Response.StatusCode = (int)responseMessage.StatusCode;
-                var responseJson = await responseMessage.Content.ReadAsStringAsync();
+
                 return Content(responseJson, "application/json");
```

Comment:
Thinking about this more (and seeing the pattern emerge), we literally are doing the exact same thing regardless of error OR success, so we don't even need the test, we can simply pass through the content 1:1 AND the status code; eliminating a bit more verbose logic here.